src/fixate/_switching.py

```diff
@@ -43,17 +43,33 @@
     Dict,
     FrozenSet,
     Iterable,
+    Literal,
+    get_origin,
+    get_args,
+    Any,
 )
 from dataclasses import dataclass
 from functools import reduce
 from operator import or_
+from types import new_class
+
+import sys
```

Comment:
Assuming this conditional import won't be necessary if we're heading straight for 3.12?